src/UserSpaceInstrumentation/InstrumentProcess.cpp

```diff
@@ -246,9 +247,13 @@ ErrorMessageOr<void> SetOrbitThreadsInTarget(pid_t pid, const std::vector<Module
   constexpr const char* kSetOrbitThreadsFunctionName = "SetOrbitThreads";
   OUTCOME_TRY(void* set_orbit_threads_function_address,
               DlsymInTracee(pid, modules, library_handle, kSetOrbitThreadsFunctionName));
-  OUTCOME_TRY(ExecuteInProcess(pid, set_orbit_threads_function_address, orbit_threads[0],
-                               orbit_threads[1], orbit_threads[2], orbit_threads[3],
-                               orbit_threads[4], orbit_threads[5]));
+  const auto safe_get_tid = [&orbit_threads](size_t index) {
+    if (index < orbit_threads.size()) return orbit_threads.at(index);
+    return -1;
+  };
+  OUTCOME_TRY(ExecuteInProcess(pid, set_orbit_threads_function_address, safe_get_tid(0),
```

Comment:
Hmm, maybe this should be better discussed with @danielfenner , but above we have `ORBIT_CHECK(orbit_threads.size() == GetExpectedOrbitThreadNames().size())`, so here we are already guaranteed to have exactly `orbit_threads.size() == 5`. It seems to me that we should just change `OrbitUserSpaceInstrumentation.cpp/SetOrbitThreads` to take five parameters (and the underlying array to have size 5), and that this was simply forgotten when updating gRPC.

In other words, making `OrbitUserSpaceInstrumentation.cpp` expect a fixed number of thread names may sound like something we want to fix, but in reality the whole injection code here relies on that number to be fixed.

Reply:
Basically, 

> The code assumes that the resulting list contains at least 6 TIDs. But that recently changed.

is not the case. The code assumes a fixed number.

Reply:
Oh yeah. You're right. This is more complicated than I thought. I'm gonna ignore this problem for now. Thanks for the review though!

Reply:
But what about just changing `OrbitUserSpaceInstrumentation.cpp/SetOrbitThreads` to take five TIDs instead of six? If the number changes again this problem could appear again, but at least it should fix the  out-of-bounds access for now.